cli/cmd/cluster.go

```diff
@@ -315,21 +321,19 @@ func internalClusterPublishModule(cmdCtx *cmdcontext.CmdCtx, args []string) erro
 	}
 
 	// It looks like an application or an application:instance.
-	instanceCtx, name, err := parseAppStr(cmdCtx, args[0])
+	configPath, appName, instName, err := parseAppStr(cmdCtx, args[0])
 	if err != nil {
 		return err
 	}
-
-	configPath := instanceCtx.ClusterConfigPath
 	if configPath == "" {
-		if name != "" {
+		if instName != "" {
 			return fmt.Errorf("can not to update an instance configuration " +
 				"if a cluster configuration file does not exist for the application")
 		}
-		configPath = filepath.Join(instanceCtx.AppDir, defaultConfigFileName)
+		instEnabledPath := running.GetInstanceEnabledPath(cliOpts, cmdCtx.Cli.ConfigDir)
+		configPath = filepath.Join(instEnabledPath, appName, defaultConfigFileName)
```

Comment:
This will work with yml/yaml correctly. Also, this will not work for instances.enabled = . and current directory is an application. There is not need to append app name in this case.

Reply:
Fixed, config.yml/config.yaml parametrizing is added to the tests, also the test with inplace instances.enabled is here now.